data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/ownership/ConfigBucketOwnerProviderFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.ownership;
+
+import com.amazon.dataprepper.plugins.source.S3SourceConfig;
+import com.amazon.dataprepper.plugins.source.SqsQueueUrl;
+
+import java.net.MalformedURLException;
+
+public class ConfigBucketOwnerProviderFactory {
+    public BucketOwnerProvider createBucketOwnerProvider(final S3SourceConfig s3SourceConfig) {
+        if(s3SourceConfig.isDisableBucketOwnershipValidation())
+            return new NoOwnershipBucketOwnerProvider();
+
+        final String queueUrl = s3SourceConfig.getSqsOptions().getSqsUrl();
+        final String accountId;
+        try {
+            accountId = SqsQueueUrl.parse(queueUrl).getAccountId();
+        } catch (final MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+
```

Comment:
Take it or leave it. Broken up variable assignment can be problematic in the future if someone alters this functionality resulting in a null `accountId`.

I would recommend extracting out getting the accountId to a separate function

```
   public BucketOwnerProvider createBucketOwnerProvider(final S3SourceConfig s3SourceConfig) {
        if(s3SourceConfig.isDisableBucketOwnershipValidation())
            return new NoOwnershipBucketOwnerProvider();
        final String accountId = getAccountId(s3SourceConfig);
        return new StaticBucketOwnerProvider(accountId);
   }

   private String getAccountId(final S3SourceConfig s3SourceConfig) {
        final String queueUrl = s3SourceConfig.getSqsOptions().getSqsUrl();

        try {
            return SqsQueueUrl.parse(queueUrl).getAccountId();
        } catch (final MalformedURLException e) {
            throw new RuntimeException(e);
        }
   }
```